### PR TITLE
Rename ADU *device_information, send update result details

### PR DESCRIPTION
**Pull request by ewertons**
## Purpose
The ADU documentation refers to the device basic information as
"device properties", not "device information", so this change renames
the structures and variables to match that naming.
Also the ADU sample gets an improvement to send a description for the
update result details, as well as an user-defined extended result code.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: demos/projects/ESPRESSIF/adu/main/azure_iot_freertos_esp32_pnp_model.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ uint32_t ulSampleCreateReportedPropertiesUpdate( uint8_t * pucPropertiesData,
 
 // Azure Device Update
 extern az_iot_adu_client_update_request xUpdateRequest;
-extern az_iot_adu_client_device_information xDeviceInformation;
+extern az_iot_adu_client_device_properties xDeviceInformation;
```
Comment (danewalton) [inline demos/projects/ESPRESSIF/adu/main/azure_iot_freertos_esp32_pnp_model.h:77]: ugh I wish this was named something else with all the overloading of the term "device properties". But it has the ADU prefix stuff so I think it's okay

Comment (danewalton) [inline demos/projects/ESPRESSIF/adu/main/azure_iot_freertos_esp32_pnp_model.h:77]:
```suggestion
extern az_iot_adu_client_device_properties xADUDeviceProperties;
```

File: demos/sample_azure_iot_adu/sample_azure_iot_adu.c [hunk 2]
```diff
@@ -577,9 +579,9 @@ static AzureIoTResult_t prvEnableImageAndResetDevice()
      * through pucResultDetails.
      */
     xUpdateResults.lResultCode = 0;
-    xUpdateResults.lExtendedResultCode = 0;
-    xUpdateResults.pucResultDetails = NULL;
-    xUpdateResults.ulResultDetailsLength = 0;
+    xUpdateResults.lExtendedResultCode = 1234;
```
Comment (danewalton) [inline demos/sample_azure_iot_adu/sample_azure_iot_adu.c:582]: does this number have some significance? If I'm a first time user I might think this means something. So maybe a macro titled `sampleaduEXAMPLE_EXTENDED_RESULT` or something to make it explicit this is just made up?

File: demos/sample_azure_iot_adu/sample_azure_iot_adu.c [hunk 3]
```diff
@@ -590,9 +592,9 @@ static AzureIoTResult_t prvEnableImageAndResetDevice()
     for( int32_t ulStepIndex = 0; ulStepIndex < xUpdateResults.ulStepResultsCount; ulStepIndex++ )
     {
         xUpdateResults.pxStepResults[ ulStepIndex ].ulResultCode = 0;
-        xUpdateResults.pxStepResults[ ulStepIndex ].ulExtendedResultCode = 0;
-        xUpdateResults.pxStepResults[ ulStepIndex ].pucResultDetails = NULL;
-        xUpdateResults.pxStepResults[ ulStepIndex ].ulResultDetailsLength = 0;
+        xUpdateResults.pxStepResults[ ulStepIndex ].ulExtendedResultCode = 5678;
```
Comment (danewalton) [inline demos/sample_azure_iot_adu/sample_azure_iot_adu.c:595]: same here

File: demos/sample_azure_iot_adu/sample_azure_iot_adu.c [hunk 4]
```diff
@@ -181,6 +181,8 @@ static uint32_t ulReportedPropertiesUpdateLength;
 
 uint8_t ucAduContextBuffer[ ADU_CONTEXT_BUFFER_SIZE ];
 
+#define sampleaduDEFAULT_RESULT_DETAILS "Ok"
```
Comment (danewalton) [inline demos/sample_azure_iot_adu/sample_azure_iot_adu.c:184]: nit: we can make this a `uint8_t` buffer.

**Review by danewalton — APPROVED** [review]
Looks good. If you could double check for any lingering values named "device information" 👍🏼

File: libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c [hunk 5]
```diff
@@ -371,7 +371,7 @@ AzureIoTResult_t AzureIoTADUClient_SendResponse( AzureIoTADUClient_t * pxAzureIo
 }
 
 static void prvFillBaseDeviceInformation( AzureIoTADUClientDeviceInformation_t * pxDeviceInformation,
```
Comment (danewalton) [inline libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c:373]: maybe do the middleware one too?

File: demos/projects/ESPRESSIF/adu/main/azure_iot_freertos_esp32_pnp_model.h (hunk 1 above)
Comment (ewertons) [reply demos/projects/ESPRESSIF/adu/main/azure_iot_freertos_esp32_pnp_model.h:77]: Yeah, I rather keep it similar to the json property defined by the ADU team.
And as you mentioned, these properties and structures have "ADU" as prefix, so it is still specific to ADU.

Comment (ewertons) [reply demos/projects/ESPRESSIF/adu/main/azure_iot_freertos_esp32_pnp_model.h:77]: Fixed.

File: libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c (hunk 5 above)
Comment (ewertons) [reply libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c:373]: Yes, there were a bunch of vars and structs left out of the first set. I updated the name (hopefully) everywhere else that was applicable.

File: demos/sample_azure_iot_adu/sample_azure_iot_adu.c (hunk 2 above)
Comment (ewertons) [reply demos/sample_azure_iot_adu/sample_azure_iot_adu.c:582]: That sounds like a good idea. I'll move it to be a macro.

File: demos/sample_azure_iot_adu/sample_azure_iot_adu.c (hunk 3 above)
Comment (ewertons) [reply demos/sample_azure_iot_adu/sample_azure_iot_adu.c:595]: That sounds like a good idea. I'll move it to be a macro.

File: demos/sample_azure_iot_adu/sample_azure_iot_adu.c (hunk 4 above)
Comment (ewertons) [reply demos/sample_azure_iot_adu/sample_azure_iot_adu.c]: done

**Pull request merged**